src/backend-api/Controllers/DeleteController.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using backend_api.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+
+namespace backend_api.Controllers
+{
+    // [Authorize]
+    [ApiController]
+    public class DeleteController : ContextController
+    {
+        public DeleteController(ITInventoryDBContext context) : base(context) { }
+
+        /* PUT: api/{operation}/{model}/{id}
+         * Will change the IsDeleted field for the id of the model corresponding to the operation.
+         * Return: 200 if updated. Else, 400 bad request. 
+         */
+        [HttpPut]
+        [Route("{operation}/{model}/{id}")]
+        public IActionResult PutDepartment([FromRoute] string operation, string model, int id)
+        {
+            model = VerbatimMatch(model);
+
+            // Try to change the operation to a boolean.
+            bool isDeleted;
+            try
+            {
+                isDeleted = this.isDeleted(operation);
+            }
+            catch (Exception)
+            {
+                return BadRequest($"Invalid operation: {operation}");
+            }
+
+            switch (model)
+            {
+                case "employee":
+                    return BadRequest("Not Deleted");
+                case "program":
+                    return BadRequest("Not Deleted");
+                case "department":
+                    return DeleteDepartment(isDeleted, id);
+                case "server":
+                    return BadRequest("Not Deleted");
+                case "computer":
+                    return BadRequest("Not Deleted");
+                case "monitor":
+                    return BadRequest("Not Deleted");
+                case "peripheral":
+                    return BadRequest("Not Deleted");
+                default:
+                    return BadRequest("Invalid Model");
+            }
+        }
+
+        /* isDeleted(operation) converts the route path string
+         *      to a boolean value or throws an error if the string
+         *      is not "delete" or "recover"
+         * Params: string operation
+         * Returns: boolean.
+         */
+        private bool isDeleted(string operation)
```

Comment:
This method seems to be checking the operation argument rather than checking if something is deleted. Maybe change method name?